include/seqan3/search/search_result.hpp

```diff
@@ -0,0 +1,200 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::search_result.
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/std/concepts>
+#include <exception>
+
+#include <seqan3/core/detail/debug_stream_type.hpp>
+#include <seqan3/core/detail/empty_type.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+
+namespace seqan3::detail
+{
+struct policy_result_builder;
+}
+
+namespace seqan3
+{
+
+/*!\brief A data holder for the search result.
+ * \ingroup search
+ * \tparam query_id_type The type of the query_id.
+ * \tparam cursor_type The type of the cursor.
+ * \tparam reference_id_type The type of the reference_id.
+ * \tparam reference_begin_pos_type The type of the reference_begin_pos.
+ *
+ * \if DEV
+ * \note If the information is not available the type of the respective data is of seqan3::detail::empty_type.
+ * \endif
+ */
+template <typename query_id_type, typename cursor_type, typename reference_id_type, typename reference_begin_pos_type>
+class search_result
```

Comment:
This class has no constraints on the types. But these would be good to know since this is also public interface in order to understand what the user can expect on the returned types.